File: Fullstack Web App/Backend Source Code/Product/Product.java
package com.example.ecommercebackend.Product;

import jakarta.persistence.*;


@Entity
@Table(name = "products")
public class Product {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "product_id")
    private int productId;

    private String name;
    private String description;
    private double price;
    private int category_id;

    private String imagepath;

    public int getId() {
        return productId;
    }

    public String getName() {
        return name;
    }

    public String getDescription() {
        return description;
    }

    public double getPrice() {
        return price;
    }

    public int getProduct_id() {
        return productId;
    }

    public int getCategory_id() {
        return category_id;
    }

    public String getImagepath() {
        return imagepath;
    }

    // Constructors, Getters, and Setters
}

File: Fullstack Web App/Frontend Source Code/product.dart
class Product {
  final int product_id;
  final String name;
  final String description;
  final double price;
  final int category_id;
  final String imagepath;

  Product({
    required this.product_id,
    required this.name,
    required this.description,
    required this.price,
    required this.imagepath,
    required this.category_id,
  });

  factory Product.fromJson(Map<String, dynamic> json) {
    print('imagepath: ${json['imagepath']}'); // Add this line

    return Product(
      product_id: json['product_id'] != null ? json['product_id'] : 0,
      name: json['name'] != null ? json['name'] : '',
      description: json['description'] != null ? json['description'] : '',
      price: json['price'] != null ? json['price'].toDouble() : 0.0,
      category_id: json['category_id'] != null ? json['category_id'] : 0,
      imagepath: json['imagepath'] != null ? json['imagepath'] : '',
    );
  }

  int get productId => product_id;
}
